inference_engine/src/controller.py

```diff
@@ -36,20 +37,30 @@ def _add_io_path(exam_path: str, io_path: str):
     @staticmethod
     def _mark_detection(file_path: str, file_name: str, output_path: str):
         image = Storage.get_exam_image(file_path, file_name)
-        answer_input, index_input = Preprocessing().process(image)
-        index_result = IndexModel(Config.paths.index_model_path).inference(index_input)
-        logging.info(f"Detected index: {index_result}")
-        answer_result = AnswerModel(Config.paths.answer_model_path).inference(answer_input)
+        fields_images = Preprocessing(image).process()
```

Comment:
Hi @marekhering as we discussed offline, could you separate pre/post-processing and inference into a separate method? e.g. something like
```
def check_image(image: PIL.Image.Image) -> (answer_results, exam_key_result, ocr_results)
```
It would make this part much more portable to other controller parts